// then-busboy/lib/cjs/Body.js
"use strict";
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
    if (kind === "m") throw new TypeError("Private method is not writable");
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
};
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var _Body_entries;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Body = void 0;
const object_deep_from_entries_1 = __importDefault(require("object-deep-from-entries"));
const formdata_node_1 = require("formdata-node");
const BodyField_1 = require("./BodyField");
const pathToFieldname_1 = __importDefault(require("./util/pathToFieldname"));
const isField_1 = __importDefault(require("./util/isField"));
const isFile_1 = __importDefault(require("./util/isFile"));
class Body {
    constructor(entries) {
        _Body_entries.set(this, void 0);
        __classPrivateFieldSet(this, _Body_entries, entries.map(([path, value]) => [
            path,
            // Convert raw entry scalar values to BodyField class
            (0, isFile_1.default)(value) || (0, isField_1.default)(value)
                ? value
                : new BodyField_1.BodyField(value, (0, pathToFieldname_1.default)(path))
        ]), "f");
    }
    /**
     * A shortcut for Body#json()
     */
    static json(value) {
        if (value instanceof Body) {
            return value.json();
        }
        return new Body(value).json();
    }
    /**
     * A shortcut for Body#formData()
     */
    static formData(value) {
        if (value instanceof Body) {
            return value.formData();
        }
        return new Body(value).formData();
    }
    get [(_Body_entries = new WeakMap(), Symbol.toStringTag)]() {
        return "Body";
    }
    /**
     * Returns an amount of entries in Body
     */
    get length() {
        return __classPrivateFieldGet(this, _Body_entries, "f").length;
    }
    *entries() {
        yield* __classPrivateFieldGet(this, _Body_entries, "f").values();
    }
    *keys() {
        for (const [key] of this) {
            yield key;
        }
    }
    *values() {
        for (const [, value] of this) {
            yield value;
        }
    }
    [Symbol.iterator]() {
        return this.entries();
    }
    /**
     * Returns a new Body subset with only files in it
     */
    files() {
        const entries = [];
        for (const [path, value] of this) {
            if ((0, isFile_1.default)(value)) {
                entries.push([path, value]);
            }
        }
        return new Body(entries);
    }
    /**
     * Returns a new Body subset with only fields in it
     */
    fields() {
        const entries = [];
        for (const [path, value] of this) {
            if (!(0, isFile_1.default)(value)) {
                entries.push([path, value]);
            }
        }
        return new Body(entries);
    }
    /**
     * Returns an object representing Body entries
     */
    json() {
        const entries = [];
        for (const [path, value] of this) {
            entries.push([path, (0, isFile_1.default)(value) ? value : value.valueOf()]);
        }
        return (0, object_deep_from_entries_1.default)(entries);
    }
    /**
     * Returns FormData representation of Body entries
     */
    formData() {
        const form = new formdata_node_1.FormData();
        for (const [path, value] of this) {
            form.append((0, pathToFieldname_1.default)(path), (0, isFile_1.default)(value) ? value : value.valueOf());
        }
        return form;
    }
}
exports.Body = Body;
