// then-busboy/lib/esm/BodyFileDataItem.js
var __classPrivateFieldSet = (this && this.__classPrivateFieldSet) || function (receiver, state, value, kind, f) {
    if (kind === "m") throw new TypeError("Private method is not writable");
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a setter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot write private member to an object whose class did not declare it");
    return (kind === "a" ? f.call(receiver, value) : f ? f.value = value : state.set(receiver, value)), value;
};
var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (receiver, state, kind, f) {
    if (kind === "a" && !f) throw new TypeError("Private accessor was defined without a getter");
    if (typeof state === "function" ? receiver !== state || !f : !state.has(receiver)) throw new TypeError("Cannot read private member from an object whose class did not declare it");
    return kind === "m" ? f : kind === "a" ? f.call(receiver) : f ? f.value : state.get(receiver);
};
var _BodyFileDataItem_file, _BodyFileDataItem_path, _BodyFileDataItem_enc;
export class BodyFileDataItem {
    constructor({ file, path, enc }) {
        _BodyFileDataItem_file.set(this, void 0);
        _BodyFileDataItem_path.set(this, void 0);
        _BodyFileDataItem_enc.set(this, void 0);
        __classPrivateFieldSet(this, _BodyFileDataItem_file, file, "f");
        __classPrivateFieldSet(this, _BodyFileDataItem_path, path, "f");
        __classPrivateFieldSet(this, _BodyFileDataItem_enc, enc, "f");
    }
    get [(_BodyFileDataItem_file = new WeakMap(), _BodyFileDataItem_path = new WeakMap(), _BodyFileDataItem_enc = new WeakMap(), Symbol.toStringTag)]() {
        return "File";
    }
    get name() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_file, "f").name;
    }
    get size() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_file, "f").size;
    }
    get type() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_file, "f").type;
    }
    get lastModified() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_file, "f").lastModified;
    }
    get path() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_path, "f");
    }
    get enc() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_enc, "f");
    }
    slice(start, end, contentType) {
        return __classPrivateFieldGet(this, _BodyFileDataItem_file, "f").slice(start, end, contentType);
    }
    stream() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_file, "f").stream();
    }
    arrayBuffer() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_file, "f").arrayBuffer();
    }
    text() {
        return __classPrivateFieldGet(this, _BodyFileDataItem_file, "f").text();
    }
}
